### DISCUSSION TITLE
Doors (and hatches) pressure resistance

### DISCUSSION THREAD

**Discussion by Stallfighter** [FakeFishGames/Barotrauma#6087] [Ideas and suggestions]
Currently pressure can damage only sub walls. Doors will be unaffected regardless how high pressure is behind it. I suggest to add pressure resistance to doors and hatches as well - airlock and dock ones should have special tag so they are getting same pressure resistance as hull ones. For internal doors however, there has to be pressure check and gradual deterioration while under pressure. You can counter that by constantly repairing them, OR weld them, so weld will deteriorate instead. Deterioration speed should be tied to overpressure level (i.e. how deep we are compared to door pressure resistance stat), and water level in compartment that door tries to protect (for example, if its half-flooded, its deterioration rate is 0.5 for given depth).
You can upgrade doors as well. There are 2 possible ways - either link door/hatch upgrades with hull upgrades, but all internal doors (without "airlock" or "dock" tags) will have 10-20% less crush depth, OR you can add entire new upgrade tree for doors, with pressure resistance, repair speed, repair skill level and welding speed. It may as well increase its resistance against alien attacks.

**Comment by Hazmat616** [comment]
I love this idea and would say voltage and general use should also cause them to deteriorate but instead of breaking open then break closed and can't open until you fix them or use a crowbar.
